File: showcase/tide.go
// Package main presents a small coastal observation.
package main

import "fmt"

type Reading struct {
    Station string
    Height  float64
}

func (r Reading) Summarize() string {
    return fmt.Sprintf("%s: %.2f m", r.Station, r.Height)
}

func main() {
    reading := Reading{Station: "North Cove", Height: 1.72}
    fmt.Println(reading.Summarize())
}

File: showcase/tide.py
"""A tide table, with a small and predictable interface."""
from dataclasses import dataclass
from datetime import datetime, timezone


@dataclass(frozen=True)
class Reading:
    station: str
    height: float
    measured_at: datetime


def summarize(readings: list[Reading]) -> str:
    if not readings:
        return "Awaiting a reading"
    latest = max(readings, key=lambda item: item.measured_at)
    return f"{latest.station}: {latest.height:.2f} m"


sample = Reading("North Cove", 1.72, datetime.now(timezone.utc))
print(summarize([sample]))
